pkg/report/datafeeds/types.go

```diff
@@ -19,26 +21,28 @@ type Reports = []FeedReport
 var schema = GetSchema()
 
 func GetSchema() abi.Arguments {
-	mustNewType := func(t string, internalType string, components []abi.ArgumentMarshaling) abi.Type {
-		result, err := abi.NewType(t, internalType, components)
+	// Helper function to simplify error handling when creating new ABI types.
+	mustNewType := func(typ string, internalType string, components []abi.ArgumentMarshaling) abi.Type {
+		result, err := abi.NewType(typ, internalType, components)
 		if err != nil {
 			panic(fmt.Sprintf("Unexpected error during abi.NewType: %s", err))
 		}
 		return result
 	}
 
 	return abi.Arguments([]abi.Argument{
-		// TODO: why is the workflow encoder_config "(bytes32 FeedID, bytes RawReport)[] Reports"?
 		{
-			Type: mustNewType("tuple(bytes32, bytes)[]", "", []abi.ArgumentMarshaling{
+			// This defines the array of tuple records.
+			Type: mustNewType("tuple(bytes32,uint224,uint32)[]", "", []abi.ArgumentMarshaling{
```

Comment:
This schema doesn't fit the above `FeedReport` type.

But also, how do you plan to support current two types of EVM schema's used (one for Celo, and another for all other EVM workflows): https://github.com/search?q=repo%3Asmartcontractkit%2Fchainlink-deployments%20encoder_config&type=code